### PR TITLE
Import/Export Slashing Protection

### PR DESCRIPTION
**Pull request by olehyev138**
Created Two components slashing protection component. for managing import or export of the slashing components. Import Protection component responsible for importing the slashing protections.

Abstracted the import component dropzone.

Created an extension file and extended the Date object with an additional method.

### CODE REVIEW

File: src/app/modules/wallet/components/import-protection/import-protection.component.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
+
+<div class="import-keys-form">
+    <div class="text-white text-xl mt-4">
+        Import Slashing protections
```
Comment (rauljordan) [inline src/app/modules/wallet/components/import-protection/import-protection.component.html:4]:
```suggestion
        Import Slashing Protection
```

File: src/app/modules/wallet/components/import-protection/import-protection.component.html [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    </div>
+    <ng-container *ngIf="!isUploading">
+        <div class="my-6 text-hint text-lg leading-relaxed">
+            Upload any folder of slashing protection files
```
Comment (rauljordan) [inline src/app/modules/wallet/components/import-protection/import-protection.component.html:8]:
```suggestion
            Upload your slashing protection file. To read more about how to obtain this file, see our documentation portal <a href="https://docs.prylabs.network/docs/wallet/slashing-protection">here</a>
```

**Comment by rauljordan** [comment]
Screen Shot 2021-03-17 at 6 24 14 PM
The text is a little cut off here. The breadcrumb should also say "slashing" instead of "slashin"

File: src/app/modules/wallet/pages/slashing-protection/slashing-protection.component.html [hunk 3]
```diff
@@ -0,0 +1,59 @@
+<app-breadcrumb></app-breadcrumb>
+<div *ngIf="!isImport" class="onboarding bg-default flex h-screen">
+    <div class="mx-auto overflow-y-auto">
+        <div class="text-center pb-8">
+            <div class="text-white text-3xl">Import/Export Slashing protection</div>
```
Comment (rkapka) [inline src/app/modules/wallet/pages/slashing-protection/slashing-protection.component.html:5]:
```suggestion
            <div class="text-white text-3xl">Import/Export Slashing Protection</div>
```

File: src/app/modules/wallet/pages/slashing-protection/slashing-protection.component.html [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            my-auto">
+            <div #import (click)="toggleActive(export,import)" class="bg-paper
+                onboarding-card text-center flex flex-col my-8
+                md:my-0">
+                <div class="wallet-info">
+                    <p class="wallet-kind">
+                        Import Slashing Protection
+                    </p>
+                    <p class="wallet-description">
+                        Import a slashing protection Json
```
Comment (rkapka) [inline src/app/modules/wallet/pages/slashing-protection/slashing-protection.component.html:18]:
```suggestion
                        Import a slashing protection JSON
```

File: src/app/modules/wallet/pages/slashing-protection/slashing-protection.component.html [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            <div (click)="toggleActive(import,export)" #export class="bg-paper
+                onboarding-card text-center flex flex-col my-8
+                mx-8
+                md:my-0 active">
+                <div class="wallet-info">
+                    <p class="wallet-kind">
+                        Export Slashing Protection
+                    </p>
+                    <p class="wallet-description">
+                        Get a slashing protection Json
```
Comment (rkapka) [inline src/app/modules/wallet/pages/slashing-protection/slashing-protection.component.html:41]:
```suggestion
                        Get a slashing protection JSON
```

File: src/app/modules/wallet/pages/slashing-protection/slashing-protection.component.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  exportSlashingProtection(): void {
+    this.walletService
+      .exportSlashingProtection()
+      .pipe(
+        tap((response) => {
+          const blob = new Blob([response.file], {
+            type: 'text/plain;charset=utf-8',
+          });
+          const d = new Date();
+          const fileName = `slashingp-protections${d.toDateTimeString()}.json`;
```
Comment (rkapka) [inline src/app/modules/wallet/pages/slashing-protection/slashing-protection.component.ts:48]:
```suggestion
          const fileName = `slashing-protection-${d.toDateTimeString()}.json`;
```

File: src/app/modules/wallet/components/import-protection/import-protection.component.ts [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+            this.isUploading = false;
+            this.notificationService.notifySuccess(
+              'Successfully imported your slashing protection'
+            );
+            this.back();
+          }),
+          catchError((error) => {
+            this.isUploading = false;
+            this.notificationService.notifyError(
+              'Error happened while import your slashing protection'
```
Comment (rkapka) [inline src/app/modules/wallet/components/import-protection/import-protection.component.ts:70]:
```suggestion
              'An error occured while importing your slashing protection'
```

**Review by rkapka — CHANGES_REQUESTED** [review]
I think there are a few things missing.

1. There is no confirmation after a successful import.
2. https://deploy-preview-170--prysm-web-ui.netlify.app/dashboard/wallet/accounts/import displays a few nice things: number of files uploaded, error when the format is wrong (although the styling is a little bit off), etc. We should have the same in slashing protection import.
3. We can probably make a common component for both slashing protection upload and the upload from the previous point, which includes all the nice things mentioned in (2), so that we can reuse it everywhere. I guess the file format should be a parameter.

File: src/app/modules/shared/components/import-accounts-form/import-accounts-form.component.html [hunk 8]
```diff
@@ -45,7 +45,9 @@
         your node is not running
       </mat-error>
     </mat-form-field>
-    <div class="my-6 flex flex-wrap">
+    <app-import-dropzone [accept]="'.json,.zip'"
+      (fileChange)="fileChangeHandler($event)"></app-import-dropzone>
+    <!-- <div class="my-6 flex flex-wrap">
```
Comment (rkapka) [inline src/app/modules/shared/components/import-accounts-form/import-accounts-form.component.html:50]: Please remove commented code.

File: src/app/modules/shared/components/import-accounts-form/import-accounts-form.component.html [hunk 9]
```diff
@@ -72,12 +74,12 @@
         Max 50 keystore files allowed. If you have more than that, we recommend
         an HD wallet instead
       </mat-error>
-      <mat-error *ngIf="invalidFiles.length">
+      <!-- <mat-error *ngIf="invalidFiles.length">
```
Comment (rkapka) [inline src/app/modules/shared/components/import-accounts-form/import-accounts-form.component.html:77]: Please remove commented code.

**Review by rkapka — CHANGES_REQUESTED** [review]
Please remove dead code.

**Review by rkapka — APPROVED** [review]

**Pull request merged**